src/test/java/org/synyx/urlaubsverwaltung/application/web/ApplicationForLeaveDetailsControllerTest.java

```diff
@@ -0,0 +1,802 @@
+
+        verify(applicationInteractionService, never()).cancel(any(), any(), any());
+    }
+
+    @Test
+    public void cancelApplicationRedirectsToApplication() throws Exception {
+
+        final Person signedInPerson = somePerson();
+        when(personService.getSignedInUser()).thenReturn(signedInPerson);
+        when(applicationService.getApplicationById(APPLICATION_ID)).thenReturn(Optional.of(applicationOfPerson(signedInPerson)));
+
+        perform(post("/web/application/" + APPLICATION_ID + "/cancel"))
+            .andExpect(status().isFound())
+            .andExpect(header().string("Location", "/web/application/" + APPLICATION_ID));
+    }
+
+    @Test
+    public void remindBossWithUnknownApplicationIdThrowsUnknownApplicationForLeaveException() {
+
+        assertThatThrownBy(() ->
+
+            perform(post("/web/application/" + APPLICATION_ID + "/remind"))
+
+        ).hasCauseInstanceOf(UnknownApplicationForLeaveException.class);
+    }
+
+    @Test
+    public void remindBossAddsFlashAttributIfThrowsNoException() throws Exception {
+
+        when(applicationService.getApplicationById(APPLICATION_ID)).thenReturn(Optional.of(someApplication()));
+
+        perform(post("/web/application/" + APPLICATION_ID + "/remind"))
+            .andExpect(flash().attribute("remindIsSent", true));
+    }
+
+    @Test
+    public void remindBossAddsFlashAttributIfThrowsRemindAlreadySentException() throws Exception {
+
+        final Application application = someApplication();
+        when(applicationService.getApplicationById(APPLICATION_ID)).thenReturn(Optional.of(application));
+        when(applicationInteractionService.remind(application)).thenThrow(RemindAlreadySentException.class);
+
+        perform(post("/web/application/" + APPLICATION_ID + "/remind"))
+            .andExpect(flash().attribute("remindAlreadySent", true));
+    }
+
+    @Test
+    public void remindBossAddsFlashAttributIfThrowsImpatientAboutApplicationForLeaveProcessException() throws Exception {
+
+        final Application application = someApplication();
+        when(applicationService.getApplicationById(APPLICATION_ID)).thenReturn(Optional.of(application));
+        when(applicationInteractionService.remind(application)).thenThrow(ImpatientAboutApplicationForLeaveProcessException.class);
+
+        perform(post("/web/application/" + APPLICATION_ID + "/remind"))
+            .andExpect(flash().attribute("remindNoWay", true));
+    }
+
+    @Test
+    public void remindBossRedirectsToApplication() throws Exception {
+
+        when(applicationService.getApplicationById(APPLICATION_ID)).thenReturn(Optional.of(someApplication()));
+
+        perform(post("/web/application/" + APPLICATION_ID + "/remind"))
+            .andExpect(status().isFound())
+            .andExpect(header().string("Location", "/web/application/" + APPLICATION_ID));
+    }
+
+    public Person somePerson() {
```

Comment:
Die folgenden Methoden bitte im scope auf `private` stellen, diese werden und sollen ja nur in dieser Test Methode verwendet werden.